### ISSUE TITLE
Work Package Template Epic

### ISSUE THREAD

**Issue by ninest** [Northeastern-Electric-Racing/FinishLine#2029]
### Overview

Allow users to select templates for work packages to auto populate/add data to the work package form/

Sample templates: https://docs.google.com/document/d/1zqvbX-py6fH7_DrcAGJwiUTCwlj4u9sDHv315-wioIw/edit

### Stakeholders

Product Stakeholder: Amrutha Hari, @parth
Software Stakeholder: @RChandler234 , @Peyton-McKee
Reference Users: Sofia Varner, Walter Stadolnik

### User Story

1. As an Admin, I want create a library of reusable work package templates which are project specific, so that the Team Heads and Project Leads can quickly create work packages without starting from scratch.
2. As a Team Head, I want to have a standardized template for specific project types, so that all team members use a consistent format and structure. 
3. As a Project Lead,  I want to be able to auto-populate certain fields in work package templates, so that I can reduce the amount of manual data entry required.
4. As a Project Team Member, I want to have clear and concise instructions, so that I can accurately fill out all the sections of work package templates without any confusion.
5. As a new Project Team Member,  I want to see examples of how to fill out certain fields in work packages, so that I can have a better understanding of the expected format.

### Success Metrics

Once users use and create work package templates.

### Rollout Plan

This feature is planned to be done by …. Eboard and heads can experiment with the feature for the next 1/2 months using the templates at https://docs.google.com/document/d/1zqvbX-py6fH7_DrcAGJwiUTCwlj4u9sDHv315-wioIw/edit

### Out of Scope

Users creating personal templates. Templates will be managed by admins (project heads/leads).

### Background / Context

Many work packages have similar details (expected activities, deliverables, blocked by). This feature will allow users to select a template while creating or editing the work package form to auto populate/add values to the form and make it easier to create similar work packages.

### Acceptance Criteria & Mock-ups

Mockups on Figma: https://www.figma.com/file/qmXuySSwfiD942HwmSSEMe/NER-Product-and-Design?type=design&node-id=5538-11885&mode=design&t=HmEtK5DC0dqbOYOH-0

New view of work package form (create and edit form):

<img width="921" alt="image" src="https://github.com/Northeastern-Electric-Racing/FinishLine/assets/34677361/5bcae060-4e97-49e7-89fb-4fc934927521">

The templates section has been added to the top of the form. When a template is selected, the data defined in the template will be appended to the form fields. No data will be overwritten.

Admin view to create work package templates:
TODO

### Tickets

 - [ ] #
 - [ ] #
